### PR TITLE
Chat changes

### PR DESCRIPTION
**Pull request by GuilhermeFaga**
- Created a Chat class and gathered chat related functions
- Added min and max width to chat history style
- Chat auto scrolling when scrolled to bottom
- Fixed chat log of new connected players
- Minor logic fixes

### CODE REVIEW

**Comment by steviegt6** [comment]
epic

File: public/scripts/model/Chat.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
+const HTML_classes = {
```
Comment (aeuu) [inline public/scripts/model/Chat.js]: I would suggest renaming this to htmlClasses to preserve consistency,

**Review by aeuu — APPROVED** [review]

**Review by aeuu — APPROVED** [review]

**Review by valkyrienyanko — APPROVED** [review]

**Review by valkyrienyanko — APPROVED** [review]

**Review by valkyrienyanko — CHANGES_REQUESTED** [review]
(1)
Observation: When resizing the chat, you go up in chat a bit.
Expected: When resizing the chat, you should always stay at the position you were looking at (e.g. bottom of chat)

(2) 
Observation: Clicking the chat history highlights it.
Expected: Clicking the chat history does not highlight it.

(3)
Observation: When client A connects everything works. When client B connects, client A does not receive a message saying client A connected.
Expected: See a message from client A when they connect from client B.

**Comment by steviegt6** [comment]
> 
> 
> (1)
> Observation: When resizing the chat, you go up in chat a bit.
> Expected: When resizing the chat, you should always stay at the position you were looking at (e.g. bottom of chat)
> 
> (2)
> Observation: Clicking the chat history highlights it.
> Expected: Clicking the chat history does not highlight it.
> 
> (3)
> Observation: When client A connects everything works. When client B connects, client A does not receive a message saying client A connected.
> Expected: See a message from client A when they connect from client B.

What do you mean they see a message when they connect from client B?

**Comment by valkyrienyanko** [comment]
No `Client name has joined` messages are sent ever. Only disconnect messages are logged.

**Comment by steviegt6** [comment]
> 
> 
> No `Client name has joined` messages are sent ever. Only disconnect messages are logged.

Are you saying that:
If client B joins AFTER client A, client A should receive a message? Or, when client B joins, client B should receive a message stating that client A joined?

Or something else?

**Comment by valkyrienyanko** [comment]
I'm saying 

> If client B joins AFTER client A, client A should receive a message.

**Review by valkyrienyanko — APPROVED** [review]

**Pull request merged**